programs/mango-v4/src/instructions/perp_place_order.rs

```diff
@@ -109,8 +110,22 @@ pub fn perp_place_order(ctx: Context<PerpPlaceOrder>, order: Order, limit: u8) -
 
     let now_ts: u64 = Clock::get()?.unix_timestamp.try_into().unwrap();
 
-    // TODO apply reduce_only flag to compute final base_lots, also process event queue
-    require!(order.reduce_only == false, MangoError::SomeError);
+    let effective_pos = account
+        .perp_position(perp_market_index)?
+        .effective_base_position_lots();
```

Comment:
imo this should take into account orders the user already has on the book. If the user has effective_pos 5 plus ask_lots = 4, they should only be able to add an ask of size 1.

It's unclear how bid_lots would affect this. Ignore them?

Reply:
hmm agreed, I have done something like this now
![image](https://user-images.githubusercontent.com/89031858/210213673-ea43edca-dfb6-4871-b373-3f2f06d62746.png)

Reply:
please check latest change